xtas/tasks/es.py

```diff
@@ -100,32 +139,58 @@ def fetch_query_batch(idx, typ, query, field='body'):
     hits = scan(_es(), {'query': query}, index=idx, doc_type=typ)
     return [hit['_source'][field] for hit in hits if field in hit['_source']]
 
-CHECKED_MAPPINGS = set()
```

Comment:
Well, but it queries ES anyway, and only adds the mapping if it's not already there. So the only thing that's being saved is a single ES mapping query per result. And I think it should probably be private, but that's issue #111 already, so I'll leave that out of this one.